sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -644,6 +692,9 @@ int64_t CurlSession::ReadFromSocket(uint8_t* buffer, int64_t bufferSize)
       }
     }
   }
+#ifdef WINDOWS
+  WinSocketSetBuffSize(this->m_curlSocket, this->m_logger);
+#endif // WINDOWS
```

Comment:
This should be before the loop, not after. It is odd that we are setting this after the call to `curl_easy_recv`.

Reply:
This was taken from how it was applied on libcurl proper usage.
See: https://github.com/curl/curl/blob/master/lib/transfer.c#L1153

This call is used only in the case when a transfer would require more than one socket write. That's why it is call at the end of the write.
See the original issue about it: https://curl.haxx.se/mail/lib-2018-07/0080.html

And PR with the fix: https://github.com/curl/curl/pull/2762/files

Actually we can't add this fix directly to libcurl, because `easy_send()` does only one write to a socket. We are implementing the loop logic to keep writing to the socket on our side. (in contrast, the libcurl proper way would handle this for us)

On the proper way, this function keeps been called right after a socket is written. We just follow same approach